Dockerfile

```diff
@@ -1,38 +1,52 @@
-FROM alpine:edge AS builder
+FROM phusion/baseimage:0.10.2 as builder
 LABEL maintainer="chevdor@gmail.com"
 LABEL description="This is the build stage for Substrate. Here we create the binary."
 
-RUN apk add build-base \
-    cmake \
-    linux-headers \
-    openssl-dev \
-    clang-dev \
-    cargo
-
 ARG PROFILE=release
 WORKDIR /substrate
 
 COPY . /substrate
 
-RUN cargo build --$PROFILE
+RUN apt-get update && \
+	apt-get upgrade -y && \
+	apt-get install -y cmake pkg-config libssl-dev git clang
+
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y && \
+	export PATH="$PATH:$HOME/.cargo/bin" && \
+	rustup toolchain install nightly && \
+	rustup target add wasm32-unknown-unknown --toolchain nightly && \
+	cargo install --git https://github.com/alexcrichton/wasm-gc && \
+	rustup default nightly && \
+ 	./scripts/build.sh && \
+ 	rustup default stable && \
+	cargo build --$PROFILE
 
 # ===== SECOND STAGE ======
 
-FROM alpine:edge
+FROM phusion/baseimage:0.10.2
```

Comment:
Could we build against musl instead, statically link everything, and then do the second stage from Alpine, or even better `FROM scratch`?

Reply:
That would be very cool indeed. I do lack time atm however and I wanted to first have a working solution. We can always build up from that point on.